core/src/main/scala/com/yahoo/maha/core/RequestModel.scala

```diff
@@ -1115,6 +1119,21 @@ object RequestModel extends Logging {
     }
   }
 
+  def validateLengthForFilterValue(publicFactOption: Option[PublicFact], publicDimOption: Option[PublicDimension], filter: Filter): Boolean = {
+    val pubCol = if (publicFactOption.isDefined) publicFactOption.get.columnsByAliasMap(filter.field) else publicDimOption.get.columnsByAliasMap(filter.field)
+    val dataType = if (publicFactOption.isDefined) publicFactOption.get.dataTypeForAlias(pubCol.alias) else publicDimOption.get.nameToDataTypeMap(pubCol.name)
+    dataType match {
+      case StrType(length, _, _) => filter match {
+        case InFilter(_, values, _, _) if length == 0 && values.forall(_.length <= MAX_ALLOWED_STR_LEN) || values.forall(_.length <= length) => true
+        case NotInFilter(_, values, _, _) if length == 0 && values.forall(_.length <= MAX_ALLOWED_STR_LEN) || values.forall(_.length <= length) => true
+        case EqualityFilter(_, value, _, _) if length == 0 && value.length <= MAX_ALLOWED_STR_LEN || value.length <= length => true
+        case NotEqualToFilter(_, value, _, _) if length == 0 && value.length <= MAX_ALLOWED_STR_LEN || value.length <= length => true
+        case LikeFilter(_, value, _, _) if length == 0 && value.length <= MAX_ALLOWED_STR_LEN || value.length <= length => true
+        case _ => false
```

Comment:
please include IsNullFilter and other filters, ideally you should have all the filters here otherwise its going to return false for rest of missing and error will be misleading.